src/App.tsx

```diff
@@ -1,24 +1,213 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { Etodos, ResponseError } from './types/enum';
+import { UserData } from './types/userData';
+import { Todo } from './types/Todo';
+import {
+  createTodo,
+  deleteTodoOnServer,
+  getTodos,
+  updateTodoProp,
+} from './api';
+import { TodoItem } from './components/Todo';
+import { TodoHeader } from './components/TodoHeader';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
 
-const USER_ID = 0;
+const user: UserData = {
+  createdAt: '2023-07-10T13:09:53.578Z',
+  email: 'gookidoo@gmail.com',
+  id: 11038,
+  name: 'Віктор Булденко',
+  phone: null,
+  updatedAt: '2023-07-10T13:09:53.578Z',
+  username: null,
+  website: null,
+};
+
+let isUncomplete = 0;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isShowFooter, setIsShowFooter] = useState<boolean>(true);
+  const [sortTodosBy, setSortTodosBy] = useState<Etodos>(Etodos.ALL);
+  const [error, setError] = useState<ResponseError>(ResponseError.NOT);
+  const [isToggleActiveTodos, setIsToggleActiveTodos] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const [creatingTodoTitle, setCreatingTodoTitle] = useState('');
+
+  const toggleTodosActive = () => {
+    const promiseList = todos.map((todo) => {
+      if (todo.completed !== isToggleActiveTodos) {
+        return updateTodoProp(todo.id, {
+          completed: isToggleActiveTodos,
+        });
+      }
+
+      return [];
+    });
+
+    setIsToggleActiveTodos(!isToggleActiveTodos);
+
+    Promise.all(promiseList).then(() => {
+      getTodos(user.id).then((todoList) => {
+        setTodos(todoList);
+        setIsShowFooter(Boolean(todoList.length));
+      }).catch(err => new Error(err.message))
+        .finally(() => setIsLoading(false));
+    });
+  };
+
+  const checkCompletedTodo = (arr: Todo[]) => {
+    isUncomplete = 0;
+    for (let i = 0; i < arr.length; i += 1) {
+      if (!arr[i].completed) {
+        isUncomplete += 1;
+      }
+    }
+  };
+
+  const deleteTodo = (id: number) => {
+    setIsLoading(true);
+
+    return deleteTodoOnServer(id)
+      .then(() => {
+        getTodos(user.id).then((todoList) => {
+          setTodos(todoList);
+          checkCompletedTodo(todoList);
+          setIsShowFooter(Boolean(todoList.length));
```

Comment:
Mostly `!!` uses to convert in boolean type
```suggestion
          setIsShowFooter(!!(todoList.length));
```

Reply:
https://mate-parttime.slack.com/archives/C03UQS59NRY/p1689857616536079

all these problems are discussed here and this option was accepted